indico/modules/events/contributions/util.py

```diff
@@ -244,6 +244,15 @@ def get_contributions_for_person(event, person, only_speakers=False):
             .all())
 
 
+def get_contributions_for_user(event, user):
+    """Get all contributions for a user in the given event."""
+    return (Contribution.query.with_parent(event)
+            .join(ContributionPersonLink, ContributionPersonLink.contribution_id == Contribution.id)
+            .join(EventPerson, EventPerson.id == ContributionPersonLink.person_id)
```

Comment:
I don't think you need to specify join criteria here - SA should be able to automatically derive them from the FKs/relationships.